packages/ckeditor5-core/src/editor/editorconfig.ts

```diff
@@ -432,9 +483,9 @@ import type Editor from './editor';
 export interface EditorConfig {
 	context?: Context;
 	extraPlugins?: Array<PluginConstructor<Editor>>;
-	initialData?: string;
+	initialData?: string | Record<string, string>;
 	language?: string | LanguageConfig;
-	placeholder?: string;
+	placeholder?: string | Record<string, string>;
```

Comment:
This is a change that I needed to make to keep the official multi-root editor API the same as what was shown in the guide (and people use) and to be consistent with the data format for multi-root editors that we already have, e.g. in CS.

Other than that, `DataController#set` already accepts data in this format.

However, this broadens the type, which prompted other changes. I am not sure about docs -- now we need to keep two snippet examples and explain it.

Maybe a better solution would be to have new architecture for editor configs:

*   `EditorConfig` -- without things that are different for multi-root editor and single-root editor,
*   `SingleRootEditorConfig` -- adds `initialData: string` and `placeholder: string`,
*   `MultiRootEditorConfig` -- adds `initialData: Record<string, string>` and same for `placeholder`.

I am not sure if this will not be even more confusing or problematic, though. For example, places that use `EditorConfig` now will have to know which type of config it is. OTOH, `initialData` and `placeholder` aren't used in many places in the code base. Maybe they are encapsulated inside the editor-related classes and maybe there will be no problem with recognizing what type of config interface is passed.